src/algokit/cli/generate.py

```diff
@@ -0,0 +1,121 @@
+import json
+import logging
+import pathlib
+import platform
+import re
+
+import algokit_client_generator
+import click
+
+from algokit.core import proc
+
+logger = logging.getLogger(__name__)
+
+
+def snake_case(s: str) -> str:
+    s = s.replace("-", " ")
+    s = re.sub(r"([A-Z]+)([A-Z][a-z])", r"\1_\2", s)
+    s = re.sub(r"([a-z\d])([A-Z])", r"\1_\2", s)
+    return re.sub(r"[-\s]", "_", s).lower()
+
+
+def format_client_name(output: pathlib.Path, application_file: pathlib.Path) -> pathlib.Path:
+    client_name = str(output).replace("%parent_dir%", snake_case(application_file.parent.name))
```

Comment:
looks like this one should maybe just take output as a str, since it is only using it as such?